src/main/kotlin/step2/calculator/Calculator.kt

```diff
@@ -2,44 +2,49 @@ package step2.calculator
 
 /**
  * 문자열 계산기
- *
- * 사용자가 입력한 문자열 값에 따라 사칙 연산을 수행할 수 있는 계산기를 구현해야 한다.
- * 문자열 계산기는 사칙 연산의 계산 우선순위가 아닌 입력 값에 따라 계산 순서가 결정된다. 즉, 수학에서는 곱셈, 나눗셈이 덧셈, 뺄셈 보다 먼저 계산해야 하지만 이를 무시한다.
- * 예를 들어 "2 + 3 * 4 / 2"와 같은 문자열을 입력할 경우 2 + 3 * 4 / 2 실행 결과인 10을 출력해야 한다.
- *
- * @since 2023/10/23
- * @author KimJungSik
  * */
 object Calculator {
 
     private const val DELIMITER = " "
 
-    enum class CalculatorSign(val signStr: String) {
-        PLUS("+"), MINUS("-"), MULTIPLY("*"), DIVIDE("/")
-    }
+    enum class CalculatorSign(val sign: String) {
+        PLUS("+"), MINUS("-"), MULTIPLY("*"), DIVIDE("/");
 
-    private fun checkError(inputStr: String?) {
+        /**
+         * 숫자 두개와 부호로 계산한 값을 반환
+         *
+         * @param num1 입력 값1
+         * @param num2 입력 값2
+         * @return 계산된 값
+         * */
+        fun calculate(num1: Int, num2: Int): Int = when (this) {
```

Comment:
열거타입에서 when절을 사용하셨군요! 
이 방법도 좋지만, 이 경우 상위 타입(CalculatorSign)이 하위 타입(PLUS, MINUS, ...)에 대해서 모두 알고 있어야 한다는 문제가 있죠! 만약 이런 함수표현식을 각각의 인스턴스가 책임지도록 한다음 좀 더 간단하게 추상화가 가능하지 않을까요?
```suggestion
        fun calculate(num1: Int, num2: Int): Int = this.calculate(num1, num2)
```

이런식으로 말이죠!